### PR TITLE
RFC: Keras categorical input.

### PR DESCRIPTION
**Pull request by tanzhenyu**
A replication from https://github.com/tensorflow/community/pull/188

### CODE REVIEW

**Comment by bhack** [comment]
Why we need to replicate RFC? It Is hard to follow duplicated threads

**Comment by tanzhenyu** [comment]
> Why we need to replicate RFC? It Is hard to follow duplicated threads

Per Francois' request. Please follow the TensorFlow's RFC, this is merely a mirrored RFC.

**Comment by tanzhenyu** [comment]
Closing this since the mirrored RFC has been implemented.
